core/sts-3.3.md

```diff
@@ -0,0 +1,189 @@
+policy advertisement having a `duration` key from the server.
+
+A duration of 0 means the policy expires immediately. This method can be used
+by servers to remove a previously set policy.
+
+### The `port` key
+
+This key indicates the port number that clients can use when connecting over
+TLS. This key's value MUST be a single port number.
+
+If the client is not already connected securely, it MUST close the insecure
+connection and reconnect securely on the stated port.
+
+Servers MUST send this key to non-securely connected clients.
+Servers SHOULD NOT send this key to securely connected clients, because
+it will be ignored.
+
+## No user recourse
+
+User recourse should not be possible if the client fails to establish a
+secure connection for any reason with a server which the client has a STS
+policy for. For example the user should not be presented with a dialog that
+allows them to proceed with the connection. A failure to establish a secure
+connection should be treated similarly to a server error that the user is
+unable to do anything about except retrying at a later time.
+
+This is to ensure that it is not possible to fool users into allowing a
+man-in-the-middle attack.
+
+## Security
+
+### STS policy injection
+
+It is possible for attackers to inject the STS policy into a plaintext
+connection. This causes the client to reconnect over TLS and either fail,
+or connect successfully over TLS but discover that the `sts` cap is not
+present.
+
+### Policy expiration
+
+It is possible that a client successfully receives the STS policy from a server
+but later on attackers begin to block all secure connection attempts from the
+client to the server until the policy expires. At that time, the client will
+revert back to a non-secure connection. Servers SHOULD advertise a long enough
+duration which makes this scenario less likely to happen.
+
+## Examples
+
+### Redirection to a secure port
+
+Server tells a client connecting non-securely to connect securely on port 6697.
+
+    Client: CAP LS 302
+    Server: CAP * LS :sts=port=6697
+
+After the exchange, the client disconnects and reconnects securely to the same
+server on port 6697 and proceeds as it normally would.
+
+### Setting STS policy
+
+Server tells a client that is already connected securely that the client must
+only use secure connections for roughly 6 months.
+
+    Client: CAP LS 302
+    Server: CAP * LS :sts=duration=15552000
+
+Until the policy expires:
+* The client will use the port it is currently connected to in the future to
+connect securely.
+* It will not make any non-secure connection attempts to the same server.
+
+### Ignoring invalid request
+
+Server tells a client that is connected non-securely that the client must
+use secure connections for roughly 6 months.
+
+    Client: CAP LS 302
+    Server: CAP * LS :sts=duration=31536000
+
+The client ignores this because it received the STS policy over a non-secure
```

Comment:
only "duration" is ignored, not presence of sts itself.
I don't want server to send different stuff on plain vs ssl connection.